src/trufflesom/interpreter/nodes/MessageSendNode.java

```diff
@@ -34,10 +37,73 @@ public static AbstractMessageSendNode createForPerformNodes(final SSymbol select
     return new UninitializedMessageSendNode(selector, NO_ARGS).initialize(coord);
   }
 
-  public static GenericMessageSendNode createGeneric(final SSymbol selector,
+  public static AbstractMessageSendNode createGenericUnary(final SSymbol selector,
+      final ExpressionNode rcvrNode, final long coord) {
+    UninitializedDispatchNode dispatch = new UninitializedDispatchNode(selector);
+    if (rcvrNode != null) {
```

Comment:
I'd check if it's null and return the "unwanted" generic path instead and swap both code blocks around to avoid nested code, personally

Reply:
Does this works? There's still the `instanceof` check that needs to hold, now? I guess it can be combined into one `if`.

Also, there's a general style of first trying all specializations, and have the generic case as a fall through.

Reply:
The `instanceof` would still be checked, I meant reducing the level of nesting by one instead, sorry ; but if that's the style used in the rest of the codebase as you said then it's better as is

Reply:
Done here: https://github.com/SOM-st/TruffleSOM/pull/122/commits/4b9eb269cbd64e4cd84e1c294d0a278b7bd5fe24